org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -140,7 +144,7 @@ public SourceAssistProcessor(PreferenceManager preferenceManager) {
 		}
 
 		// Generate Constructors
-		Optional<Either<Command, CodeAction>> generateConstructors = getGenerateConstructorsAction(params, context, type);
+		generateConstructors = getGenerateConstructorsAction(params, context, type, JavaCodeActionKind.SOURCE_GENERATE_CONSTRUCTORS);

```

Comment:
we're basically recomputing the actions a 2nd time, seems suboptimal. Couldn't we just copy the ones we have already and change their kind?

Reply:
Fixed.